### PR TITLE
Fix/order to user attribution

### PR DESCRIPTION
**Pull request by axelazaid**
# Changes
* updated `snowplow` and `facebook` package versions
  - updating the snowplow package resolves fan-out on the `session_id` within the `snowplow_sessions` package model and downstream models where `session_id` must be unique
 
* the `order_attribution` model was its own model
  - deleted this file and moved the attribution calculations logic to `fct_snowplow_sessions`
  - aggregated order attribution to user attribution by only getting the visitor activity up to a customer's first order date

* broke up `snowplow_sessions_gclid` (renamed to `snowplow_sessions__gclids`) into two models:
  - new model: `adwords_gclid_mapping` pulls from `adwords_click_performance` and `adwords_criteria_performance` and joins the two tables to get the `gclid`

# New
* new model: `snowplow_sessions__gclids` pull from `snowplow_sessions` and `adwords_gclid_mapping` to map the 'gclid' to snowplow sessions with ads that contain `gclid`

* new model: `snowplow_sessions__mapped` joins `snowplow_sessions__gclids` back to all snowplow sessions

* new model: `snowplow_sessions__with_channels` buckets the ads into channels
  - this model then finally gets joined into `fct_snowplow_sessions` to incorporate with the attribution logic

* added tests to all new models: ran and passed

### CODE REVIEW

File: dbt_project.yml [hunk 1]
```diff
@@ -10,31 +10,33 @@ macro-paths: ["macros"]
...
-    
+
     perfect_keto:
         vars:
-            warehouse_name: 'LOADING'   
-        
+            warehouse_name: 'LOADING'
+
     facebook_ads:
           vars:
```
Comment (eogilvy12) [inline dbt_project.yml:28]: i could be wrong but doesn't this need a `router` specified?

File: models/staging/adwords/adwords_gclid_mapping.sql [hunk 2]
```diff
@@ -0,0 +1,30 @@
+with clicks as (
+
```
Comment (eogilvy12) [inline models/staging/adwords/adwords_gclid_mapping.sql:9]: this model needs a test

File: models/marts/web_events/intermediate/snowplow_sessions__gclids.sql [hunk 3]
```diff
@@ -0,0 +1,42 @@
+with sessions as (
+
+    select * from {{ref('snowplow_sessions')}}
```
Comment (eogilvy12) [inline models/marts/web_events/intermediate/snowplow_sessions__gclids.sql:3]: this model needs a test

**Review by eogilvy12 — APPROVED** [review]
there are a few new models that need tests added but otherwise this looks great!

please hold off on merging until we pair on incorporating these changes into the existing mode dashboards

File: dbt_project.yml (hunk 1 above)
Comment (axelazaid) [reply dbt_project.yml:28]: nope!

File: models/staging/adwords/adwords_gclid_mapping.sql (hunk 2 above)
Comment (axelazaid) [reply models/staging/adwords/adwords_gclid_mapping.sql:9]: there are 4000+ null `gclids`

File: models/marts/web_events/intermediate/snowplow_sessions__gclids.sql (hunk 3 above)
Comment (axelazaid) [reply models/marts/web_events/intermediate/snowplow_sessions__gclids.sql:3]: a visitor can have more than one `session_index` so the `user_snowplow_domain_id` is going to fan out according to how many session "indices" they have in this particular model

File: models/staging/adwords/adwords_gclid_mapping.sql (hunk 2 above)
Comment (eogilvy12) [reply models/staging/adwords/adwords_gclid_mapping.sql:9]: this is fine - we did discover that the four columns (criteria_id, ad_group_id, adwords_campaign, gclid) can create a unique_id if we want this down the road

File: models/marts/web_events/intermediate/snowplow_sessions__gclids.sql (hunk 3 above)
Comment (eogilvy12) [reply models/marts/web_events/intermediate/snowplow_sessions__gclids.sql:3]: lets put a test on this for surrogate key: user_snowplow_domain_id, session_index

File: models/marts/web_events/fct_snowplow_sessions.sql [hunk 4]
```diff
@@ -5,106 +5,90 @@
...
     select
     
         sessions.*,
         date_trunc('day', cast(session_start as date)) as session_day,
-        shopify_customers.customer_created_at,
-        shopify_customers.customer_age_days,
-        shopify_customers.customer_type,
-        shopify_customers.lifetime_revenue,
-        shopify_customers.lifetime_completed_orders,
+        shopify_customers.customer_id,
```
Comment (eogilvy12) [inline models/marts/web_events/fct_snowplow_sessions.sql]: we need to be consistent about what "rules" the account - customer_id or email. then, all joins and partitions across the account should be on that field

**Review by eogilvy12 — COMMENTED** [review]
I think this is close to push - the one piece we should look at is mapping adwords in ad_performance

**Review by eogilvy12 — APPROVED** [review]
this looks good to me!

**Pull request merged**